### ISSUE TITLE
Singleton can't be dumped error in Rails 7.0.1 

### ISSUE THREAD

**Issue by ryanb** [wallaby-rails/wallaby-rails#202]
Upgrading from Rails 7.0.0 to Rails 7.0.1 caused this error when I visit a wallaby resource index page.

```
Failure/Error: Time.use_zone(find_time_zone, &block)
  ActionView::Template::Error:
  singleton can't be dumped
./app/controllers/application_controller.rb:81:in `set_time_zone!'
```

I'm uncertain yet if this is specific to my app or if it's an issue in Wallaby.

cc @tian-im

**Comment by tian-im** [comment]
Hi @ryanb, Do you have the full error stack? Did you run into this error on every wallaby resource index page or just specific page?

**Comment by ryanb** [comment]
@tian-im here you go. I think it happens on every resource index page.

```
ActionView::Template::Error:
  singleton can't be dumped
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-core-0.2.2/lib/utils/wallaby/utils.rb:10:in `dump'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-core-0.2.2/lib/utils/wallaby/utils.rb:10:in `clone'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-core-0.2.2/lib/services/wallaby/sorting/next_builder.rb:25:in `next_params'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-core-0.2.2/lib/services/wallaby/sorting/link_builder.rb:48:in `build'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-6.1.3/app/views/wallaby/resources/index.html.erb:24:in `block (2 levels) in ___sers_rbates__rbenv_versions_______lib_ruby_gems_______gems_wallaby_______app_views_wallaby_resources_index_html_erb__1693576075747762860_261780'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/helpers/capture_helper.rb:45:in `block in capture'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/helpers/capture_helper.rb:209:in `with_output_buffer'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/helpers/capture_helper.rb:45:in `capture'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/helpers/tag_helper.rb:338:in `content_tag'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-6.1.3/app/views/wallaby/resources/index.html.erb:23:in `block in ___sers_rbates__rbenv_versions_______lib_ruby_gems_______gems_wallaby_______app_views_wallaby_resources_index_html_erb__1693576075747762860_261780'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-6.1.3/app/views/wallaby/resources/index.html.erb:22:in `each'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-6.1.3/app/views/wallaby/resources/index.html.erb:22:in `___sers_rbates__rbenv_versions_______lib_ruby_gems_______gems_wallaby_______app_views_wallaby_resources_index_html_erb__1693576075747762860_261780'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/base.rb:244:in `public_send'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/base.rb:244:in `_run'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/template.rb:157:in `block in render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:208:in `instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/template.rb:361:in `instrument_render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/template.rb:155:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/template_renderer.rb:65:in `block (2 levels) in render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `block in instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/template_renderer.rb:60:in `block in render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/template_renderer.rb:75:in `block in render_with_layout'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `block in instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/template_renderer.rb:74:in `render_with_layout'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/template_renderer.rb:59:in `render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/template_renderer.rb:11:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/renderer.rb:61:in `render_template_to_object'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/renderer.rb:29:in `render_to_object'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/rendering.rb:117:in `block in _render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/base.rb:270:in `in_rendering_context'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/rendering.rb:116:in `_render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/streaming.rb:216:in `_render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/rendering.rb:103:in `render_to_body'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/rendering.rb:46:in `render_to_body'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/renderers.rb:142:in `render_to_body'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/abstract_controller/rendering.rb:25:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/rendering.rb:30:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/instrumentation.rb:22:in `block (2 levels) in render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/2.7.0/benchmark.rb:308:in `realtime'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/core_ext/benchmark.rb:14:in `ms'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/instrumentation.rb:22:in `block in render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/instrumentation.rb:91:in `cleanup_view_runtime'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activerecord-7.0.1/lib/active_record/railties/controller_runtime.rb:39:in `cleanup_view_runtime'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/instrumentation.rb:21:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/remotipart-1.4.4/lib/remotipart/render_overrides.rb:23:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/responders-3.0.1/lib/action_controller/responder.rb:240:in `default_render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/responders-3.0.1/lib/action_controller/responder.rb:172:in `to_html'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/responders-3.0.1/lib/responders/flash_responder.rb:109:in `to_html'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/responders-3.0.1/lib/action_controller/responder.rb:165:in `respond'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/responders-3.0.1/lib/action_controller/responder.rb:158:in `call'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/responders-3.0.1/lib/action_controller/respond_with.rb:213:in `respond_with'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-core-0.2.2/lib/concerns/wallaby/resources_concern.rb:376:in `index'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/basic_implicit_render.rb:6:in `send_action'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/abstract_controller/base.rb:214:in `process_action'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/rendering.rb:53:in `process_action'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/abstract_controller/callbacks.rb:234:in `block in process_action'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:118:in `block in run_callbacks'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/core_ext/time/zones.rb:66:in `use_zone'
# ./app/controllers/application_controller.rb:81:in `set_time_zone!'
# ...
```

**Comment by ryanb** [comment]
@tian-im looks like it happens when cloning params. Is there a reason you don't use `deep_dup`?

**Comment by tian-im** [comment]
@ryanb Yeah, there was a reason, the `deep_dup` method wasn't able to deep clone a hash object that contains ActiveRecord class, the class object becomes nil after deep dupped. And the class object is needed by Wallaby. Not sure if Rails has fixed it or not. I can change back to use `deep_dup` and see if it breaks the tests or not.

**Comment by ryanb** [comment]
@tian-im any update on this? It's blocking upgrading to Rails 7.0.1 at the moment.

**Comment by tian-im** [comment]
@ryanb sorry for the late reply. I've done the work [here](https://github.com/wallaby-rails/wallaby-core/pull/55) to move to use Github Actions for the CI build, and it seems like the `deep_dup` is working fine. There are a lot of changes on the main branch which I'm not planning to do a release now, but I will do a patch release so that you can continue with your upgrade.

Will keep you posted.

**Comment by tian-im** [comment]
https://github.com/wallaby-rails/wallaby-core/pull/55

**Comment by ryanb** [comment]
Thanks Tian!

**Comment by tian-im** [comment]
@ryanb wallaby-core 0.2.3 is out, could you give it a try and see if it works for you?

see https://github.com/wallaby-rails/wallaby-core/releases/tag/0.2.3

**Comment by ryanb** [comment]
@tian-im I'm unable to upgrade because [wallaby-active_record](https://rubygems.org/gems/wallaby-active_record) depends on wallaby-core 0.2.2. Perhaps loosen that dependency so it doesn't care about the patch version?

**Comment by tian-im** [comment]
@ryanb done, changed to use `wallaby-core ~> 0.2.3`. Could you give it a go?

**Comment by ryanb** [comment]
@tian-im now there's a similar issue with `wallaby` gem depending on `wallaby-active_record = 0.2.6`. Perhaps loosen that dependency too?

**Comment by tian-im** [comment]
😭  I should have loosened it

**Comment by tian-im** [comment]
@ryanb done, changed to use `wallaby-active_record ~> 0.2.7`. Could you give it a go?

**Comment by ryanb** [comment]
@tian-im that seemed to have resolved the singleton error. Seeing another one though.

```
ActionView::Template::Error:
  undefined method `new' for nil:NilClass
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-core-0.2.3/lib/concerns/wallaby/authorizable.rb:94:in `authorizer_of'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-core-0.2.3/lib/concerns/wallaby/authorizable.rb:73:in `authorized?'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-core-0.2.3/lib/concerns/wallaby/authorizable.rb:83:in `unauthorized?'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-core-0.2.3/lib/helpers/wallaby/links_helper.rb:42:in `new_link'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-6.1.4/app/views/wallaby/resources/show/_has_many.html.erb:9:in `___sers_rbates__rbenv_versions_______lib_ruby_gems_______gems_wallaby_______app_views_wallaby_resources_show__has_many_html_erb___1174892162419864626_263880'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/base.rb:244:in `public_send'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/base.rb:244:in `_run'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/template.rb:157:in `block in render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:208:in `instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/template.rb:361:in `instrument_render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/template.rb:155:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/partial_renderer.rb:251:in `block in render_partial_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `block in instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/partial_renderer.rb:246:in `render_partial_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/partial_renderer.rb:237:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/renderer.rb:81:in `render_partial_to_object'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/renderer.rb:53:in `render_partial'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/helpers/rendering_helper.rb:45:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-core-0.2.3/lib/services/wallaby/type_renderer.rb:16:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-core-0.2.3/lib/helpers/wallaby/resources_helper.rb:23:in `type_render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-6.1.4/app/views/wallaby/resources/show.html.erb:17:in `block in ___sers_rbates__rbenv_versions_______lib_ruby_gems_______gems_wallaby_______app_views_wallaby_resources_show_html_erb__4107096358573394172_263740'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-6.1.4/app/views/wallaby/resources/show.html.erb:14:in `each'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-6.1.4/app/views/wallaby/resources/show.html.erb:14:in `___sers_rbates__rbenv_versions_______lib_ruby_gems_______gems_wallaby_______app_views_wallaby_resources_show_html_erb__4107096358573394172_263740'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/base.rb:244:in `public_send'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/base.rb:244:in `_run'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/template.rb:157:in `block in render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:208:in `instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/template.rb:361:in `instrument_render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/template.rb:155:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/template_renderer.rb:65:in `block (2 levels) in render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `block in instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/template_renderer.rb:60:in `block in render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/template_renderer.rb:75:in `block in render_with_layout'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `block in instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `instrument'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/template_renderer.rb:74:in `render_with_layout'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/template_renderer.rb:59:in `render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/template_renderer.rb:11:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/renderer.rb:61:in `render_template_to_object'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/renderer/renderer.rb:29:in `render_to_object'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/rendering.rb:117:in `block in _render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/base.rb:270:in `in_rendering_context'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/rendering.rb:116:in `_render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/streaming.rb:216:in `_render_template'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionview-7.0.1/lib/action_view/rendering.rb:103:in `render_to_body'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/rendering.rb:46:in `render_to_body'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/renderers.rb:142:in `render_to_body'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/abstract_controller/rendering.rb:25:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/rendering.rb:30:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/instrumentation.rb:22:in `block (2 levels) in render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/2.7.0/benchmark.rb:308:in `realtime'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/core_ext/benchmark.rb:14:in `ms'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/instrumentation.rb:22:in `block in render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/instrumentation.rb:91:in `cleanup_view_runtime'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activerecord-7.0.1/lib/active_record/railties/controller_runtime.rb:39:in `cleanup_view_runtime'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/instrumentation.rb:21:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/remotipart-1.4.4/lib/remotipart/render_overrides.rb:23:in `render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/responders-3.0.1/lib/action_controller/responder.rb:240:in `default_render'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/responders-3.0.1/lib/action_controller/responder.rb:172:in `to_html'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/responders-3.0.1/lib/responders/flash_responder.rb:109:in `to_html'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/responders-3.0.1/lib/action_controller/responder.rb:165:in `respond'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/responders-3.0.1/lib/action_controller/responder.rb:158:in `call'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/responders-3.0.1/lib/action_controller/respond_with.rb:213:in `respond_with'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/wallaby-core-0.2.3/lib/concerns/wallaby/resources_concern.rb:399:in `show'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/basic_implicit_render.rb:6:in `send_action'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/abstract_controller/base.rb:214:in `process_action'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/action_controller/metal/rendering.rb:53:in `process_action'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/actionpack-7.0.1/lib/abstract_controller/callbacks.rb:234:in `block in process_action'
# /Users/rbates/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:118:in `block in run_callbacks'
# ...
```

Let me know if you want me to post this in a separate issue.

I also see some missing translations. Not sure why Rails 7.0.1 isn't picking up all the locale files.

Are you able to get Wallaby running in Rails 7.0.1?

**Comment by tian-im** [comment]
I haven't tried that. Let me spend some time this weekend to make sure it works in Rails 7.0.1.

**Comment by tian-im** [comment]
@ryanb yeah, using `deep_dup` isn't a proper solution either. I've fixed it in `wallaby-core` `0.2.4` and tested it with Rails 7.0.1, it works fine for me. Let me know if you get it working for you.

**Comment by ryanb** [comment]
I think this is resolved now, thanks Tian. I'm having trouble running the spec on Travis CI but I think that's a separate issue.
